File: solutions/go/raindrops/1/raindrops.go
package raindrops

import "fmt"

func Convert(number int) (drops string) {
	if number%3 == 0 {
		drops += "Pling"
	}
	if number%5 == 0 {
		drops += "Plang"
	}
	if number%7 == 0 {
		drops += "Plong"
	}
	if drops == "" {
		drops = fmt.Sprint(number)
	}
	return
}

File: solutions/haskell/raindrops/1/src/Raindrops.hs
module Raindrops (convert) where

convert :: Int -> String
convert n = ("Pling" // 3 ++ "Plang" // 5 ++ "Plong" // 7) ?: show n
  where
    a // b = if n `mod` b == 0 then a else ""
    a ?: b = case a of [] -> b; _ -> a
